tests/pytests/test_aggregate.py

```diff
@@ -1225,3 +1225,67 @@ def testWithKNN(env):
                                 'REDUCE', 'COUNT', '0', 'AS', 'c', 'SORTBY', '1', '@n',
                                 'PARAMS', '2', 'blob', create_np_array_typed([0] * dim).tobytes(), 'DIALECT', dialect)
         env.assertEqual(res[1:], expected_res)
+
+
+def setup_missing_values_index(index_missing):
+    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    conn = getConnectionByEnv(env)
+    schema = ['tag', 'TAG', 'INDEXMISSING' if index_missing else None, 'num1', 'NUMERIC', 'num2', 'NUMERIC']
+    schema = [part for part in schema if part is not None]
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', *schema).ok()
+
+    # Add some documents, with\without the indexed fields.
+    conn.execute_command('HSET', 'doc1', 'tag', 'val', 'num2', '5.5')
+    conn.execute_command('HSET', 'doc2', 'tag', 'val', 'num1', '3')
+    if index_missing:
+        conn.execute_command('HSET', 'doc3', 'num1', '3', 'num2', '2.7')
+    return env
+
+def test_aggregate_filter_on_missing_values():
+    env = setup_missing_values_index(False)
+    # Search for the documents with the indexed fields (sanity)
+    # document doc1 has no value for num1, so we expect to receive the mentioned error
+    (env.expect('FT.AGGREGATE', 'idx', '@tag:{val}', 'LOAD', '1', 'num1', 'FILTER', '@num1 > 2').error().
+     contains('num1: has no value, consider using EXISTS if applicable'))
```

Comment:
Once you have a potential result (`doc2` in this case) it may be returned before the error is raised. In this case, in non-strict timeout mode (default), you get the result back instead of the error.
You need to either use the strict timeout mode (`ON_TIMEOUT FAIL`) which will make sure no error was thrown throughout the pipeline execution for ALL results, or have an empty result set to make sure this test isn't flaky.

This is applicable to the other tests as well. I'm a bit leaning towards the option of having an empty result set.